src/PreCICE.jl

```diff
@@ -1332,5 +1340,159 @@ function actionReadIterationCheckpoint()
     return unsafe_string(msgCstring)
 end
 
+@doc """
+
+    isGradientDataRequired(dataID::Integer)::Bool
+        
+Checks if the given data set requires gradient data. We check if the data object has been intialized with the gradient flag.
+# Arguments
+- `dataID::Integer`: ID of the data to be checked. Obtained by [`getDataID`](@ref).
+
+"""
+function isGradientDataRequired(dataID::Integer)
+        return ccall((:precicec_isGradientDataRequired, "libprecice"), Cint, (Cint,), dataID)
+end
+
+@doc """
+
+    writeBlockVectorGradientData(dataID::Integer, valueIndices::AbstractArray{Cint}, gradientValues::AbstractArray{Float64})
+
+
+Write gradient data of a vector data as a block, the value of a specified vertices to a dataID.
+
+# Arguments
+- `dataID::Integer`: ID of the data to be written. Obtained by [`getDataID`](@ref).
+- `valueIndices::AbstractArray{Cint}`: Indices of the vertex.
+- `gradientValues::AbstractArray{Float64}`: The gradient values to write.
+
+# Notes
+
+Previous calls:
+ - [`initialize`](@ref) 
+
+# Examples
+
+Write gradient values of a vector data for a 2D problem with 2 vertices:
+
+```julia
+data_id = 1
+valueIndices = [1,2,3]
+gradientValues = [1.0, 2.0, 3.0, 4.0, 5.0, 6.0, 7.0, 8.0]
+writeBlockVectorGradientData(data_id, valueIndices, gradientValue)
+```
+
+"""
+function writeBlockVectorGradientData(dataID::Integer, valueIndices::AbstractArray{Cint}, gradientValues::AbstractArray{Float64})
+    _size, dimensions = size(gradientValues)
+    @assert dimensions == getDimensions()*getDimensions() "Dimensions of vector data in write_block_vector_gradient_data does not match with dimensions in problem definition. Provided dimensions: $dimensions, expected dimensions: $(getDimensions()*getDimensions())"
+
+    ccall((:precicec_writeBlockVectorGradientData, "libprecice"), Cvoid, (Cint, Cint, Ref{Cint}, Ref{Cdouble}), dataID, _size, valueIndices, gradientValues)
+end
+
+@doc """
+
+    writeScalarGradientData(dataID::Integer, valueIndex::Integer, gradientValues::AbstractArray{Float64})
+
+Write gradient data of a scalar data, the value of a specified vertex to a dataID.
+
+# Arguments
+- `dataID::Integer`: ID of the data to be written. Obtained by [`getDataID`](@ref).
+- `valueIndex::Integer`: Indice of the vertex.
+- `gradientValues::AbstractArray{Float64}`: The gradient values to write.
+
+# Notes
+
+Previous calls:
+ - [`initialize`](@ref) 
+
+# Examples
+
+Write gradient values of a scalar data for a 3D problem with 5 vertices:
+
+```julia
+data_id = 1
+vertex_id = 5
+gradientValues = [1.0 2.0 3.0]
+writeScalarGradientData(data_id, vertex_id, gradientValue)
+```
+
+"""
+function writeScalarGradientData(dataID::Integer, valueIndex::Integer, gradientValues::AbstractArray{Float64})
+    _size, dimensions = size(gradientValues)
```

Comment:
Here we only have a list for one vertex. At least that is what the example shows, Julia expects two outputs but only gets the length of the vector

Reply:
I did not get it.

Reply:
`size` returns only one value, but this notation wants to split it up into two. I suggest, remove `_size` here, as you don't need it for `writeScalarGradientData`
```julia-repl
julia> size([1,2,3])
(3,)

julia> _size,dimensions=size([1,2,3])
ERROR: BoundsError: attempt to access Tuple{Int64} at index [2]
```